Code/Framework/AzNetworking/AzNetworking/ConnectionLayer/ConnectionMetrics.inl

```diff
@@ -40,4 +40,33 @@ namespace AzNetworking
     {
         *this = ConnectionMetrics();
     }
+
+    inline void ConnectionMetrics::LogPacketSent(uint32_t byteCount, AZ::TimeMs currentTimeMs)
+    {
+        if (byteCount > 0)
+        {
+            m_packetsSent++;
+        }
+        m_sendDatarate.LogPacket(byteCount, currentTimeMs);
```

Comment:
Why not put this in the if as well?

Reply:
The log packet call on the send datarate structure with 0 bytes ticks the send rate even if nothing was transmitted.. Otherwise when traffic drops to nothing, we would never update the send rate, and the send rate would never decrease. It would just be stuck at its last value until we sent more data.